src/arrayMethodJoin.js

```diff
@@ -4,8 +4,21 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function (separator = ',') {
     // write code here
+    let joinMethod = '';
+
+    for (let idx = 0; idx < this.length; idx++) {
```

Comment:
naming is awkward
- `joinMethod` - i could think it's an actual method, but it's not - it's a `joinResult` | `resultString` etc., but clearly not a `Method`
- `idx` - not a common convention, use `i | j`